From 4421d95cf92b6080c54d0e1a9dacea29ba91368c Mon Sep 17 00:00:00 2001
From: Sean Paul <seanpaul@chromium.org>
Date: Wed, 14 May 2014 12:16:41 -0400
Subject: [PATCH 01/18] drm: Make sure we limit the disabled plane search to
 crtc

This patch fixes a small bug in the primary plane disable helper where
it intended to search planes attached to the current crtc, but it ended
up searching the entire plane list. This resulted in failure when there
was another crtc with an active plane.

BUG=chromium:336809
TEST=Tested on snow & peppy

Change-Id: I96f39cdef71260bd7f87c0f7eb1b0c9f309ced62
Signed-off-by: Sean Paul <seanpaul@chromium.org>
---
 drivers/gpu/drm/drm_plane_helper.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_plane_helper.c b/drivers/gpu/drm/drm_plane_helper.c
index e768d35..8213a5a 100644
--- a/drivers/gpu/drm/drm_plane_helper.c
+++ b/drivers/gpu/drm/drm_plane_helper.c
@@ -228,11 +228,15 @@ int drm_primary_helper_disable(struct drm_plane *plane)
 		/* Already disabled */
 		return 0;
 
-	list_for_each_entry(p, &plane->dev->mode_config.plane_list, head)
+	list_for_each_entry(p, &plane->dev->mode_config.plane_list, head) {
+		if (p->crtc != plane->crtc)
+			continue;
+
 		if (p != plane && p->fb) {
 			DRM_DEBUG_KMS("Cannot disable primary plane while other planes are still active on CRTC.\n");
 			return -EBUSY;
 		}
+	}
 
 	/*
 	 * N.B.  We call set_config() directly here rather than
-- 
1.8.3.2

